python/pyspark/sql/connect/session.py

```diff
@@ -325,7 +325,7 @@ def active(cls) -> "SparkSession":
 
     active.__doc__ = PySparkSession.active.__doc__
 
-    def table(self, tableName: str) -> DataFrame:
+    def table(self, tableName: str) -> ParentDataFrame:
```

Comment:
this was the way MyPy least complained IIRC. Let me take a look again ..

Reply:
Seems like the arguments cannot be more specific type, and return types can't be wider types (https://mypy.readthedocs.io/en/stable/common_issues.html#incompatible-overrides). So it complains about the argument.

Let me just keep them all as parent dataframe for simplicity because those types aren't user-facing anyway.

Reply:
Here is one example of the error:

```
python/pyspark/sql/classic/dataframe.py:276: error: Argument 1 of "exceptAll" is incompatible with supertype "DataFrame"; supertype defines the argument type as "DataFrame"  [override]
```